src/ast.rs

```diff
@@ -0,0 +1,547 @@
+//! Frontend-agnostic Hash abstract syntax tree type definitions.
+#![allow(dead_code)]
```

Comment:
This is temporary?

Reply:
Yes, cause otherwise it will complain that we aren't using any of the items.